src/main/java/java/lang/String.java

```diff
@@ -4294,11 +4306,29 @@ public static String valueOf(double d) {
      * @return  a string that has the same contents as this string, but is
      *          guaranteed to be from a pool of unique strings.
      *
-     * @diffblue.noSupport
+     * @diffblue.limitedSupport literal strings and string-valued constant
+     * expressions are not interned.
      */
+    // DIFFBLUE MODEL LIBRARY
     // public native String intern();
     public String intern() {
-        CProver.notModelled();
-        return CProver.nondetWithoutNullForNotModelled();
+        // Initialize the pool if needed
+        if(cproverInternPool == null) {
+            int capacity = CProver.nondetInt();
+            CProver.assume(capacity > 0);
+            cproverInternPool = new String[capacity];
```

Comment:
:question: Why isn't this done as part of the initialisation of `cproverInternPool`? Is this to avoid the static initialiser getting to complicated? If that is not a problem, it would be good to avoid this null check.